internal/db/schema/migrations/postgres/17/05_preferred_endpoints.up.sql

```diff
@@ -0,0 +1,56 @@
+begin;
+
+create table host_set_preferred_endpoint (
+  create_time wt_timestamp,
+  host_set_id wt_public_id not null
+    constraint host_set_id_fkey
+    references host_set(public_id)
+    on delete cascade
+    on update cascade,
+  priority int not null
+    check(priority > 0),
+  condition text not null
+    check(length(condition) > 0),
+  primary key(host_set_id, priority),
+  unique(host_set_id, condition)
+);
+
+-- host_set_immutable_preferred_endpoint() ensures that endpoint conditions
+-- assigned to host sets are immutable.
+create or replace function
```

Comment:
```suggestion
create function
```
Avoid `create or replace` statements. They can hide errors.